source/README.adoc

```diff
@@ -0,0 +1,171 @@
+
+:toc:
+:toc-placement!:
+= RabbitMQ Source
+
+The RabbitMQ source translates messages on a RabbitMQ exchange to CloudEvents,
+which can then be used with Knative Eventing over HTTP. The source can bind to
+an existing RabbitMQ exchange, or create a new exchange if required.
+
+toc::[]
+
+
+== Published events
+
+All messages received by the source are published with the following schema:
+
+.Event attributes
+|===
+| Attribute | Value | Notes
+
+| `type` | `dev.knative.rabbitmq.event` |
+| `source` | `/apis/v1/namespace/*$NS*/rabbitmqsources/*$NAME*#*$TOPIC*`
+   | `NS`, `NAME` and `TOPIC` are derived from the source configuration
+| `id` | A unique ID | This uses the `MessageId` if available, and a UUID otherwise
+| `subject` | The ID of the message | Empty string if no message ID is present
+| `datacontenttype` | `application/json` | Currently static
+| `key` | The ID of the message | Empty string if no message ID is present
+|===
+
+The payload of the event is set to the data content of the message.
+
+=== Samples
+
+For a message published with the payload "Hello rabbitmq!", for example with
+https://www.rabbitmq.com/management-cli.html[`rabbitmqadmin`]:
```

Comment:
what does it mean here "with `rabbitmqadmin`"?

Reply:
"Using the `rabbitmqadmin` tool to publish a message."

You can also send the same message by writing python, java, etc code, but having a CLI makes it easy to test out if things are working.